src/go/k8s/controllers/redpanda/cluster_controller.go

```diff
@@ -264,7 +260,16 @@ func (r *ClusterReconciler) Reconcile(
 		log.Info(requeueErr.Error())
 		return ctrl.Result{RequeueAfter: requeueErr.RequeueAfter}, nil
 	}
-	return ctrl.Result{}, err
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	// setting license should be at the last part as it requires AdminAPI to be running
+	if err := r.setLicense(ctx, &redpandaCluster, log); err != nil {
```

Comment:
is the call idempotent? I mean what happens if you re-set the same license? Is redpanda fine with it or is it trying to store the same license over and over?

Reply:
I looked into this and we can't check I think if same license is present as we can't decode the one referenced in secret